root/static/scripts/common/utility/reactTextContent.js

```diff
@@ -0,0 +1,25 @@
+// This file is part of MusicBrainz, the open internet music database.
+// Copyright (C) 2015 MetaBrainz Foundation
+// Licensed under the GPL version 2, or (at your option) any later version:
+// http://www.gnu.org/licenses/gpl-2.0.txt
+
+const nonEmpty = require('./nonEmpty');
+
+function reactTextContent(reactElement, previous = '') {
```

Comment:
This can take something like `<span>foo<span>bar</span></span>` and return `foobar`. It's used by the `EntityLink` component to determine if there's a name variation in the content. Right now the TT hacks around this by stripping the HTML.